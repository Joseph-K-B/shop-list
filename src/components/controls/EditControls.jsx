import React, { useState } from "react";
import css from '../list/ShopList.css'

function EditControls({item, onChange, onDelete}) {
    const[displayUpdate, setDisplayUpdate] = useState(false);

    return (
        <>
        <div>
            <input 
                type='checkbox'
                className={css.check} 
                value={item.done} 
                onChange={(e) => 
                    {onChange({...item, done: e.target.checked})
                }}
            />
        </div>
        <div>
            <button 
                type='button' 
                aria-label='open delete form' 
                onClick={() => onDelete(item.id)}
            >
                Remove
            </button>
        </div>
        {displayUpdate ? (
            <>
                <input 
                    type='text'
                    value={item.name} 
                    aria-label='update item input' 
                    onChange={(e) => {
                        onChange({
                            ...item,
                            name: e.target.value
                        })
                    }}/>
                <button onClick={() => setDisplayUpdate(false)} aria-label='apply update'>Update</button>
            </>
        ) : (            
            <button aria-label='open update form' onClick={() => setDisplayUpdate(true)}>Edit</button>
        )}
        </>
    )
}

export default EditControls;