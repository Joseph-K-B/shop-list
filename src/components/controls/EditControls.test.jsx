import React from 'react';
import { screen, render } from '@testing-library/react';
import EditControls from './EditControls';

it('renders controls to edit item', () => {
    const {container} = render(
        <EditControls item={{id: 1, name: 'test', done: false}}/>
    );

    const editBtn = screen.getByLabelText(/open update form/i);
    const deleteBtn = screen.getByLabelText(/open delete form/i);

    expect(editBtn).toBeInTheDocument();
    expect(deleteBtn).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})