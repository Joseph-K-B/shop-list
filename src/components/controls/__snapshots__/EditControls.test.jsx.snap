// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders controls to edit item 1`] = `
<div>
  <input
    type="checkbox"
    value="false"
  />
  <button
    aria-label="open delete form"
    type="button"
  >
    Remove
  </button>
  <button
    aria-label="open update form"
  >
    Edit
  </button>
</div>
`;
